clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -265,9 +259,68 @@ public FSDataOutputStream append(Path path, int i, Progressable progressable) th
     }
 
     @Override
-    public boolean rename(Path path, Path path1) throws IOException {
-        LOG.debug("$$$$$$$$$$$$$$$$$$$$$$$$$$$$ rename $$$$$$$$$$$$$$$$$$$$$$$$$$$$ ");
-        return false;
+    public boolean rename(Path src, Path dst) throws IOException {
+        LOG.debug("$$$$$$$$$$$$$$$$$$$$$$$$$$$$ Rename path {} to {} $$$$$$$$$$$$$$$$$$$$$$$$$$$$", src, dst);
+
+        // This implementation supports rename when src and dst are files. directories are not supported. TODO (Tals): support dirs
+
+        FileStatus srcStatus;
+        try {
+            srcStatus = getFileStatus(src);
+        } catch (FileNotFoundException e) {
+            LOG.error("rename: src not found {}", src);
+            return false;
+        }
+
+        ObjectLocation srcObjectLoc = pathToObjectLocation(src);
+        ObjectLocation dstObjectLoc = pathToObjectLocation(dst);
+        if (srcObjectLoc.equals(dstObjectLoc)) {
+            LOG.debug("rename: src and dst refer to the same lakefs object location: {}", dst);
+            return srcStatus.isFile();
+        }
+
+        ObjectStats stats;
+        try {
+            // Stat src file to get its metadata
+            stats = getObjectStats(src);
+        } catch (ApiException e) {
+            LOG.error("Rename: Could not get src Object stats. src:{}", src, e);
+            return false;
+        }
+
+        StagingApi staging = new StagingApi(apiClient);
+        StagingLocation srcStagingLoc;
+        try {
+            srcStagingLoc = staging.getPhysicalAddress(srcObjectLoc.getRepository(),
```

Comment:
I don't understand: this requests a physical path _where we could write a new file_.  I don't think we need it here if we're doing just data operations.  The existing src file is in any case specified on `srcStatus`.

Reply:
You are right, I don't need this here. I chose to use the StageObject endpoint and removed this code.